### ISSUE TITLE
Relative imports incorrectly calculated

### ISSUE THREAD

**Issue by mangelozzi** [PyCQA/pylint#3073]
# Steps to reproduce
src\a\b\c\one.py
```
foo = 1
```

src\a\b\c\two.py
```
from .one import foo
```
Result: E0401: Unable to import 'b.c..one.foo'

echo $PYTHONPATH
```
...;c:\full\path\to\a\b;...
```

### Current behavior
Result: E0401: Unable to import 'b.c..one.foo'

### Expected behavior
Try import 'b.c.one.foo' not 'b.c..one.foo'
(Notice the double dot)

### pylint --version output
pylint 2.3.1
astroid 2.2.5
Python 3.7.4 (tags/v3.7.4:e09359112e, Jul  8 2019, 20:34:20) [MSC v.1916 64 bit (AMD64)]

### Notes
I looked at other bug reports but they mostly deal with adding the required dir to the syspath, which is not the case here.

**Comment by PCManticore** [comment]
Thanks for the report!

**Comment by bsiem** [comment]
Not sure if this is the same issue, but I see something similar after upgrading pylint 2.4.4 to pylint 2.5.2, using Python 3.8.

Having a package `b` in directory `a` with three files:

`a/b/__init__.py`:
<pre>from .one import ONE
from .two import TWO</pre>

`a/b/one.py`:
<pre>ONE = 1</pre>

`a/b/two.py`:
<pre>from .one import ONE
TWO = ONE + ONE</pre>

pylint 2.4.4 + astroid 2.3.3 do not generate any warning or error, which is as expected.

pylint 2.5.2 + astroid 2.4.1 generates:
<pre>$ export PYTHONPATH=/full/path/to/a
$ cd /full/path/to
$ pylint a
************* Module a.b
a/b/__init__.py:2:0: E0401: Unable to import 'a.b.two' (import-error)</pre>

Versions:
<pre>$ pylint --version
pylint 2.5.2
astroid 2.4.1
Python 3.8.3 (default, May 17 2020, 18:15:42) 
[GCC 10.1.0]</pre>

**Comment by clavedeluna** [comment]
I've run both examples provided and am not able to reproduce any issue related to import-error, even while playing a bit with `PYTHONPATH`.

If anyone can reproduce, please reopen.
